rfcs/select-component.md

```diff
@@ -154,15 +154,17 @@ export default () => {
   Disable control from being changed
 * `required: boolean`:
   Mark control as required
+* `tabindex: number`:
+  The start offset of tabindex to navigate through component. Default is `0`.  
 * `overrides: {}`
   * `DropDown: ?React.ComponentType` component to use for dropdown list
   * `Option: ?React.ComponentType` component to use for options in dropdown list
   * `Root: ?React.ComponentType` component to use for most top of the select component
   * `Input: ?React.ComponentType` component for Input showing current selected value(s). See `Input` Control of this framework for reference to override it's functionality.
   * `SearchIcon: ?React.ComponentType` component for all icons appearing in Select component. It's provided `$type: ICON.loop | ICON.clearTag | ICON.clearAll | ICON.selected` to setup corresponding icon of Select component
   * `Tag: ?React.ComponentType` component for selected options Tags shown in Input for multiple mode selection
-* `onChange: func(e: SyntheticInputEvent, params: Object)`:
-  handler for events on trigger element when option are changing selection or text of search input (in Search mode has changed). `params` has `type` indicating which action is performed, `id` and `label` of selected\unselected option and `selectedOptions` array of all of selected, has new `textValue` set in input (for `keyUp`)
+* `onChange: func(e: SyntheticInputEvent, params: Object) => Promise`:
```

Comment:
Why does this need to be a Promise?

Reply:
because when we want to update Options if text value has changed (in Search mode for instance), than it might be async call to request options from server.

Reply:
Talked with Oleg and we came up with some proposed changes that reduce the complexity of this change handler:
https://github.com/uber-web/baseui/issues/171#issuecomment-424884154